File: Answer/118.杨辉三角.cpp
/*
 * @lc app=leetcode.cn id=118 lang=cpp
 *
 * [118] 杨辉三角
 */

// @lc code=start
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
		if (numRows == 0)
		{
			vector<vector<int>> n(0);
			return n;
		}
		vector<vector<int>> n(numRows);
		for (int i = 0; i < numRows; i++)
		{
            n[i] = vector<int>(i + 1);
			for (int j = 0; j < i + 1; j++)
			{
				if (j == 0 || j == i) n[i][j] = 1;
				else
				{
					n[i][j] = n[i - 1][j - 1] + n[i - 1][j];
				}
			}
		}
		return n;
    }
};
// @lc code=end

File: 118.杨辉三角.cs
/*
 * @lc app=leetcode.cn id=118 lang=csharp
 *
 * [118] 杨辉三角
 */

// @lc code=start
public class Solution
{
    public IList<IList<int>> Generate(int numRows)
    {
        var res = new List<IList<int>>();
        res.Add(new List<int> { 1 });
        for (int i = 1; i < numRows; i++)
        {
            var row = new List<int>();
            row.Add(1);
            for (int j = 1; j < i; j++)
            {
                row.Add(res[i - 1][j - 1] + res[i - 1][j]);
            }
            row.Add(1);
            res.Add(row);
        }
        return res;
    }
}
// @lc code=end

